src/core/cdr/src/dds_cdrstream.c

```diff
@@ -4429,10 +4377,440 @@ uint32_t dds_stream_type_nesting_depth (const uint32_t * __restrict ops)
   return nesting_depth;
 }
 
+
+
+
+// Calculate key size
+
+static uint32_t add_to_key_size_impl (uint32_t keysize, uint32_t field_size, uint32_t field_dims, uint32_t field_align, uint32_t max_align)
+{
+  uint32_t sz = keysize;
+  if (field_align > max_align)
+    field_align = max_align;
+  if (sz % field_align)
+    sz += field_align - (sz % field_align);
+  sz += field_size * field_dims;
+  if (sz > DDS_FIXED_KEY_MAX_SIZE)
+    sz = DDS_FIXED_KEY_MAX_SIZE + 1;
+  return sz;
+}
+
+static void add_to_key_size_xcdrv1 (struct key_props *k, uint32_t field_size, uint32_t field_dims, uint32_t field_align)
+{
+  if (k->min_xcdrv == DDSI_RTPS_CDR_ENC_VERSION_1)
```

Comment:
You're right, there was already a check in `add_to_key_size` (which I think is still useful because it avoids some calculations if v1 key size is not applicable). Fixed in [54e2bfc](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1750/commits/54e2bfc67e74c28a984ad4d3dbbac0920f33a35b)